python/lsst/sims/utils/CompoundCoordinateTransformations.py

```diff
@@ -0,0 +1,493 @@
+    and parallactic angle using PALPY
+
+    @param [in] raRad is RA in radians.  Can be a numpy array or a single value.
+    Assumed to be in the International Celestial Reference System.
+
+    @param [in] decRad is Dec in radians.  Can be a numpy array or a single value.
+    Assumed to be in the International Celestial Reference System.
+
+    @param [in] obs is an ObservationMetaData characterizing
+    the site of the telescope and the MJD of the observation
+
+    @param [out] altitude in radians
+
+    @param [out] azimuth in radians
+
+    @param [out] parallactic angle in radians
+    """
+
+    # TODO time this should be passing around Universal Time
+
+    raIsArray = False
+    decIsArray = False
+    if isinstance(raRad, np.ndarray):
+        raIsArray = True
+
+    if isinstance(decRad, np.ndarray):
+        decIsArray = True
+
+    if raIsArray and not decIsArray:
+        raise RuntimeError('passed numpy array of RA to altAzPaFromRaDec; but only one Dec')
+
+    if decIsArray and not raIsArray:
+        raise RuntimeError('passed numpy array of Dec to altAzPaFromRaDec; but only one RA')
+
+    if raIsArray and decIsArray and len(raRad) != len(decRad):
+        raise RuntimeError('in altAzPaFromRaDec length of RA numpy array does not match length of Dec numpy array')
+
+
+    if not hasattr(raRad, '__len__'):
+        raObs_temp, decObs_temp = _observedFromICRS(np.array([raRad]), np.array([decRad]), obs_metadata=obs,
+                                                      includeRefraction=True, epoch=2000.0)
+
+        raObs = raObs_temp[0]
+        decObs = decObs_temp[0]
+
+
+    else:
+        raObs, decObs = _observedFromICRS(raRad, decRad, obs_metadata=obs, epoch=2000.0, includeRefraction=True)
+
+    lst = calcLmstLast(obs.mjd.UT1, obs.site.longitude)
+    last = lst[1]
+    haRad = np.radians(last*15.0) - raObs
+
+
+    #sfd time This should be passing around Universal Time
+
+    if isinstance(haRad, np.ndarray):
+        az, azd, azdd, \
+        alt, altd, altdd, \
+        pa, pad, padd = palpy.altazVector(haRad, decObs, obs.site.latitude)
+    else:
+        az, azd, azdd, \
+        alt, altd, altdd, \
+        pa, pad, padd = palpy.altaz(haRad, decObs, obs.site.latitude)
+
+    return alt, az, pa
+
+
+def raDecFromAltAz(alt, az, obs):
+    """
+    Convert altitude and azimuth to RA and Dec
+
+    @param [in] alt is the altitude in degrees.  Can be a numpy array or a single value.
+
+    @param [in] az is the azimuth in degrees.  Cant be a numpy array or a single value.
+
+    @param [in] obs is an ObservationMetaData characterizing
+    the site of the telescope and the MJD of the observation
+
+    @param [out] RA in degrees (in the International Celestial Reference System)
+
+    @param [out] Dec in degrees (in the International Celestial Reference System)
+
+    Note: This method is only accurate to within 0.01 arcsec near azimuth = 0 or pi
```

Comment:
Actually, since this was already here, if this is discussed in a previous pull request, never mind.